task/host-sp-comms/src/bsp/gimlet_bcd.rs

```diff
@@ -0,0 +1,674 @@
+                self.tx_buf.try_encode_inventory(sequence, b"U615/ID", || {
+                    let InventoryData::VpdIdentity(identity) = &mut data
+                        else { unreachable!(); };
+                    *identity = packrat
+                        .get_identity()
+                        .map_err(|_| InventoryDataResult::DeviceAbsent)?
+                        .into();
+                    Ok(&data)
+                });
+            }
+            17 => {
+                // U615: Gimlet VPD EEPROM
+                //
+                // Note that for VPD AT24CSW080 identities, we allocate our
+                // InventoryData in the outer frame then pass it in as a
+                // reference; `read_at24csw080_id` typically isn't inlined, and
+                // we're already paying a stack frame for the data in this
+                // function, so it saves us 512 bytes of stack.
+                let mut data = InventoryData::At24csw08xSerial([0u8; 16]);
+                self.read_at24csw080_id(
+                    sequence,
+                    b"U615",
+                    i2c_config::devices::at24csw080_local_vpd,
+                    &mut data,
+                )
+            }
+            18 => {
+                // J180/ID: Fan VPD barcode (not available in packrat)
+                self.read_fan_barcodes(
+                    sequence,
+                    b"J180/ID",
+                    i2c_config::devices::at24csw080_fan_vpd,
+                )
+            }
+            19 => {
+                // J180: Fan VPD EEPROM
+                let mut data = InventoryData::At24csw08xSerial([0u8; 16]);
+                self.read_at24csw080_id(
+                    sequence,
+                    b"J180",
+                    i2c_config::devices::at24csw080_fan_vpd,
+                    &mut data,
+                )
+            }
+            // Welcome to The Sharkfin Zone
+            //
+            // Each Sharkfin has 3 inventory items:
+            // - Oxide barcode
+            // - Raw VPD EEPROM ID register
+            // - Hot-swap controller
+            //
+            // Sharkfin connectors start at J206 and are numbered sequentially
+            20..=29 => {
+                let (designator, f) =
+                    Self::get_sharkfin_vpd(index as usize - 20);
+                let mut name = *b"____/U7/ID";
+                name[0..4].copy_from_slice(&designator);
+                self.read_eeprom_barcode(sequence, &name, f)
+            }
+            30..=39 => {
+                let (designator, f) =
+                    Self::get_sharkfin_vpd(index as usize - 30);
+                let mut name = *b"____/U7";
+                name[0..4].copy_from_slice(&designator);
+                let mut data = InventoryData::At24csw08xSerial([0u8; 16]);
+                self.read_at24csw080_id(sequence, &name, f, &mut data)
+            }
+            // TODO: Sharkfin HSC?
```

Comment:
"do you want to address this before merging" check

Reply:
Not in this PR, no; the HSCs will be relevant for the future PR that adds `SensorId` to all of our types